src/Stormpath/Cache/Cache.php

```diff
@@ -35,5 +35,12 @@ public function delete($key);
      */
     public function clear();
 
+    /**
+     * Get all items from cache
+     *
+     * @return array
+     */
+    public function all();
```

Comment:
We shouldn't need to have this method because the problem seems to be that we are storing the Href + the query string as the key of some resources in the cache. This would also force people who decide to implement (or that already have) their own cacheing mechanism to implement this method. The fix should really be to only cache the resources by Href, excluding any query string. A reason for this is that we currently don't cache collection resources so retrieving a resource with a query string for search or expansion of collection resources will very likely return an inaccurate result.

However, I think this is a good time to implement cacheing of the individual resources returned on a search or collection resource expansion. See how this is implemented in the [Java SDK for reference](https://github.com/stormpath/stormpath-sdk-java/blob/master/impl/src/main/java/com/stormpath/sdk/impl/ds/DefaultDataStore.java#L759).

**Note**: only cacheing of api key would require to cache Href + query string, but that's not yet implemented in the PHP SDK.